### DISCUSSION TITLE
Does ArgoCD support AWS S3?

### DISCUSSION THREAD

**Discussion by koala-dev-123** [argoproj/argo-cd#8091] [Q&A] [answered]
I'm thinking of using ArgoCD for deploying to AWS EKS and using S3 as a storage location for Manifest files.
Does ArgoCD support S3?

**Comment by alexmt** [comment, marked as answer]
S3 is not supported. Currently, only Git and Helm repositories are supported. The main reason is that Argo CD needs something that supports versioning.

**Reply by koala-dev-123** [reply]
I'm happy with your answer. I would like to use CodeCommit.
Thank.
Does ArgoCD use S3 with Helm repository? 
[ArgoCD helm plugin](https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-plugins)

**Reply by yuriy-yarosh** [reply]
@alexmt Modern S3 does support versioning.

This question is quite relevant for IaC because storing intermediate artifacts inside the Gitops repo is not really a viable option, i.e. in terms of Terraform IaC it usually goes like this:

Terraform IaC -> Applying Charts -> Generating Kustomize -> (Stored Somewhere) -> ArgoCD picking it up.
It's often necessary to propagate some IaC governed resources for the target app, right now there's no clear way for that.

Crossplane is not flexible and reliable enough for IaC nowadays, as for me, personally... although would fit perfectly in such scenario as TF replacement.
